stdlib/array.ncl

```diff
@@ -271,7 +254,7 @@
         "%m
       = fun f n => %generate% n f,
 
-    sort | forall a. (a -> a -> ComparisonResult) -> Array a -> Array a
+    sort | forall a. (a -> a -> [| `Lesser, `Equal, `Greater |]) -> Array a -> Array a
```

Comment:
Would it be desirable to make this accessible to lib consumers? If someone wants to define a sorting function and statically type it, wouldn't they need to spell out the enum type?

But I suppose this would require some kind of static type aliases.

Reply:
Yes, I initially kept an alias for this, but then the typechecker doesn't understand as an enum type statically. If we have type alias at some point (and actually contract equality may end up doing that with a slight enhancement) this may be a better API. For now, I fear we have no choice, if we want static typing.